cmd/support-profile.go

```diff
@@ -32,13 +64,175 @@ var supportProfileCmd = cli.Command{
+	inputFile, err := os.Open(sourcePath)
+	if err != nil {
+		return err
+	}
+
+	outputFile, err := os.Create(destPath)
+	if err != nil {
+		inputFile.Close()
+		return err
+	}
+	defer outputFile.Close()
+
+	_, err = io.Copy(outputFile, inputFile)
+	inputFile.Close()
+	if err != nil {
+		return err
+	}
+
+	// The copy was successful, so now delete the original file
+	return os.Remove(sourcePath)
+}
+
+func getProfileData(client *madmin.AdminClient) string {
+	// Create profile zip file
+	tmpFile, e := ioutil.TempFile("", "mc-profile-")
+	fatalIf(probe.NewError(e), "Unable to download profile data.")
+
+	// Ask for profile data, which will come compressed with zip format
+	zippedData, adminErr := client.DownloadProfilingData(globalContext)
+	fatalIf(probe.NewError(adminErr), "Unable to download profile data.")
+
+	// Copy zip content to target download file
+	_, e = io.Copy(tmpFile, zippedData)
+	fatalIf(probe.NewError(e), "Unable to download profile data.")
+
+	// Close everything
+	zippedData.Close()
+	tmpFile.Close()
+
+	downloadPath := "profile.zip"
+	downloadedFile := downloadPath + "." + time.Now().Format(dateTimeFormatFilename)
+
+	fi, e := os.Stat(downloadPath)
+	if e == nil && !fi.IsDir() {
+		e = moveFile(downloadPath, downloadedFile)
+		fatalIf(probe.NewError(e), "Unable to create a backup of profile.zip")
+	} else {
+		if !os.IsNotExist(e) {
+			fatal(probe.NewError(e), "Unable to download profile data.")
+		}
+	}
+	fatalIf(probe.NewError(moveFile(tmpFile.Name(), downloadPath)), "Unable to download profile data.")
+	return downloadedFile
 }
 
 // mainSupportProfile is the handle for "mc support profile" command.
 func mainSupportProfile(ctx *cli.Context) error {
-	commandNotFound(ctx, supportProfileSubcommands)
+	// Check for command syntax
+	checkAdminProfileSyntax(ctx)
+	// Get the alias parameter from cli
+	args := ctx.Args()
+	aliasedURL := args.Get(0)
+
+	profilers := ctx.String("type")
+	count := ctx.Int("count")
+	interval := ctx.Int("interval")
+
+	// Create a new MinIO Admin Client
+	client, err := newAdminClient(aliasedURL)
+	if err != nil {
+		fatalIf(err.Trace(aliasedURL), "Unable to initialize admin client.")
+		return nil
+	}
+
+	for i := 1; i <= count; i++ {
+		_, cmdErr := client.StartProfiling(globalContext, madmin.ProfilerType(profilers))
+		fatalIf(probe.NewError(cmdErr), "Unable to start profile. ")
+		console.Infof("count %d : Profile data successfully started. \n", i)
+
+		log.Println("Collecting data for 10 seconds")
+		sleep := time.Duration(10)
+		time.Sleep(time.Second * sleep)
+		log.Println("Stopping profiling..")
+
+		getProfileData(client)
+		console.Infof("count %d : Profile data successfully downloaded as %s\n", i, getProfileData(client))
+
+		if count > 1 && i <= count-1 {
+			log.Printf("Waiting for %d seconds, before starting another profile\n", interval)
+			sleepBetweenIntervals := time.Duration(interval)
+			time.Sleep(time.Second * sleepBetweenIntervals)
+		}
+	}
```

Comment:
Do you know that we already have `madmin.Profile()` to do this?

Reply:
Thanks replaced  `StartProfiling` and `DownloadProfilingData` with `madmin.Profile()` . But the ask here is to have a count and interval . 
```
mc admin profile --count --interval
```
It  downloads profile file based on number of times specified in the count and sleep for the interval specified.

Refer https://docs.google.com/spreadsheets/d/1JbIi1JZCF04sH6MZ5X9zr6iCNYb2c4I8sISRNMguj6c/edit#gid=0

Reply:
the count is not that useful IMO // cc @klauspost - we only ever care about is `interval` -- we have no reason to take 10 different profiles at various intervals in succession.

Reply:
@harshavardhana 99% agree, but I kind of assumed you guys had discussed it already.

The 1% would be to grab goroutines at intervals. But mostly I don't see the point.

Reply:
cc @kannappanr

Reply:
Yeah only Goroutines make sense rest have little to no value in capturing in this manner

Reply:
The point of profiling is that it samples itself, we don't have to take many samples of profiles to compare. Interval is the sampler here - not the number of times we took the profile.

Reply:
with count and interval, we will know when to stop and how many times to capture the profile

Reply:
yeah but its not going to be useful @kannappanr - it's a waste of capturing profiles.  all we care is `interval` not `count`

Reply:
@kannappanr If agreed then i can remove count and interval flag. The only flag which will be present will be `duration : start profiling for the specified duration in seconds` . This duration is the sleep time which we specify between start and stop profilers.

Reply:
AFAICS count is meaningless count is nothing but `count * duration` should be actual interval to be sent to the server, since sampling ratio is server-side for profiling, not client-side.

In a single sitting, no one will wait for 10 different profiles for the same interval.  There will be better data captured if we did `count * duration` instead